cluster.go

```diff
@@ -28,94 +40,119 @@ type ClusterClient struct {
 
 // NewClusterClient initializes a new cluster-aware client using given options.
 // A list of seed addresses must be provided.
-func NewClusterClient(opt *ClusterOptions) (*ClusterClient, error) {
-	addrs, err := opt.getAddrSet()
-	if err != nil {
-		return nil, err
-	}
-
+func NewClusterClient(opt *ClusterOptions) *ClusterClient {
 	client := &ClusterClient{
-		addrs:   addrs,
-		conns:   make(map[string]*Client),
+		addrs:   opt.getAddrs(),
+		clients: make(map[string]*Client),
 		opt:     opt,
 		_reload: 1,
 	}
 	client.commandable.process = client.process
-	client.reloadIfDue()
-	return client, nil
+	return client
 }
 
-// Close closes the cluster connection
+// Close closes the cluster client.
 func (c *ClusterClient) Close() error {
-	c.slotsMx.Lock()
-	defer c.slotsMx.Unlock()
-
-	return c.reset()
+	// TODO: close should make client unusable
+	return nil
 }
 
 // ------------------------------------------------------------------------
 
-// Finds the current master address for a given hash slot
-func (c *ClusterClient) getMasterAddrBySlot(hashSlot int) string {
-	if addrs := c.slots[hashSlot]; len(addrs) > 0 {
-		return addrs[0]
+// getClient returns a Client for a given address.
+func (c *ClusterClient) getClient(addr string) *Client {
+	c.clientsMx.RLock()
+	client, ok := c.clients[addr]
+	if ok {
+		c.clientsMx.RUnlock()
+		return client
 	}
-	return ""
-}
+	c.clientsMx.RUnlock()
 
-// Returns a node's client for a given address
-func (c *ClusterClient) getNodeClientByAddr(addr string) *Client {
-	c.connsMx.Lock()
-	client, ok := c.conns[addr]
+	c.clientsMx.Lock()
```

Comment:
There's (technically) a little race condition here, but I doubt it will actually matter